src/vcml/models/generic/fbdev.cpp

```diff
@@ -20,6 +20,162 @@ void fbdev::update() {
     }
 }
 
+template <typename T>
+void write_to_ostream(ostream& os, const T* val) {
+    os.write(reinterpret_cast<const char*>(val), sizeof(T));
+}
+
+struct bmp_header {
+    array<char, 2> header; // 0x42 0x4D for BMP (BM)
+    u32 size;              // total size (bytes)
+    u16 reserved_0;
+    u16 reserved_1;
+    u32 offset; // offset of the pixel array
+
+    u32 header_size() { return 14; }
+
+    bmp_header():
+        header({ 'B', 'M' }),
+        size(0),
+        reserved_0(0),
+        reserved_1(0),
+        offset(0) {}
+
+    void write(ostream& out) {
```

Comment:
this can be replaced by
```
ostream& operator<<(ostream& os, const bmp_header& h) {
    os << h.head[0];
    os << h.head[1];
    os << h.size;
    ...
    return os;
}